Reviving Fire Fixes and Tweaks PR.

Sindorman opened the pull request:
This PR revives Burger's PR #5797 

- Fire extinguishing foam now works properly. It is 3 times stronger than water.

- Walking through a foam will extenguish your fire stacks.

- Atmos suit max temperature threshold was increased from 30k to 40k. Atmos firefighter suit max temperature was increased from 30k to 45k. Heavy firesuit max temperature was increased from 30k to 55k. Same goes for their helmets.

- Firesuits and red/white/atmos hardhats no longer protect wearer from pressure. So people cannot use it in space. For areas with 2000+ KPa fire people need to use atmos voidsuit.

- Fixes #867

- Cyborg stations now refill cyborg's fire extinguishers. Fixes #6171 

- Fire now produces CO2 while burning, with amount varying by fire severity

Arrow768 commented on code/game/objects/effects/decals/Cleanable/fuel.dm:
```diff
@@ -154,4 +154,12 @@
\ No newline at end of file
+			amount *= 0.75
+
+/obj/effect/decal/cleanable/foam/Crossed(AM as mob|obj)
```
Not too sure if that is a good idea since that could just result in people walking in circles on extinguishing foam.
Imho it would more sense to implement that in the drop and roll thing you can do when on fire (make it more effective if you do it on a tile covered in foam

Sindorman replied:
I thought about same thing. But I want to keep this along with drop and roll. As irrc foam is thick where you would definitely get yourself covered in it by just walking. I think I will make drop and roll suggestion, decease effectiveness of walking lower then drop and roll. What do you think?

Arrow768 replied:
The problem I see with just walking through foam is that it is a very passive action, that isnt really disrupting. (Which I believe it should be)

Where as drop and roll is a rather active and disrupting action.
You pretty much have to stop doing whatever you are doing to take care of the fire.

skull132 commented on code/modules/mob/living/carbon/resist.dm:
```diff
@@ -1,22 +1,24 @@
 	//drop && roll
 	if(on_fire && !buckled)
-		fire_stacks -= 1.2
+		var/obj/effect/decal/cleanable/foam/extinguisher_foam = locate() in src.loc
```
I suspect this is bad. Shouldn't we make extinguisher foam itself process() and apply its effects?

Sindorman replied:
Arrow was against foam passively extinguish.

skull132 replied:
I'm trying to decide which is a better approach in terms of OOP. I guess it doesn't matter in the end, though.

skull132 commented on code/modules/mob/living/living_defense.dm:
```diff
@@ -257,24 +257,41 @@
 
-/mob/living/proc/ExtinguishMob()
-	if(on_fire)
+	adjust_fire_stacks(-fire_stacks_to_remove)
```
```suggestion
	if (fire_starts_to_remove)
		adjust_fire_stacks(-fire_stacks_to_remove)
```

skull132 commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -33,5 +31,6 @@
 /obj/item/clothing/head/hardhat/red/atmos
 	name = "atmospheric firefighter helmet"
 	desc = "An atmospheric firefighter's helmet, able to keep the user protected from heat and fire."
+	max_heat_protection_temperature = FIRE_HELMET_MAX_HEAT_PROTECTION_TEMPERATURE + 15000
```
This kind of use of defines kinda overrules their purpose. Consider creating more macros and redefining the max?

Sindorman replied:
Why? Fire suit is a base case and we got 2-3 exceptions.

skull132 replied:
~~The point of defines is defeated if they're generally turned into a magic number generator.~~

Sindorman commented:
Sorry for pushing commit after it is approved for merge. TLDR of commits: I have replaced firesuit sprite with better sprite that has gloves, and updated changelog for it.
